### DISCUSSION TITLE
Setting innerText of SuperElements

### DISCUSSION THREAD

**Discussion by sandy-garrido** [ulixee/secret-agent#234] [Q&A] [answered]
How could I achieve setting an element's innerText which is a hidden field?
I'm trying to overcome captcha's in particular hcaptcha, which the last part is setting a hidden field.

the obvious, isn't possible as it's only a Getter
```
const responseField = await agent.document.querySelector("[name='h-captcha-response']");
responseField.innerText = token;
```
`TypeError: Cannot set property innerText of #<SuperElement> which has only a getter`

ps, sorry for bombarding the team/community with queries - I'm back to work tomorrow :joy:

**Comment by blakebyrnes** [comment, marked as answer]
We don't have a great answer for this right now. We'll add setters to awaited-dom now that they've allowed setters to have different signatures than getters, but it's not a small effort. There's a hacky interim solution to just use agent.getJsValue(..) and provide a string that sets the value.

**Reply by sandy-garrido** [reply]
That actually worked :) Thanks!
